password/src/Data/Password/Validate.hs

```diff
@@ -0,0 +1,243 @@
+{-# LANGUAGE CPP               #-}
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+{-|
+Module      : Data.Password.Valid
+Copyright   : (c) Dennis Gosnell, 2019; Felix Paulusma, 2020; Hiroto Shioi, 2020
+License     : BSD-style (see LICENSE file)
+Maintainer  : cdep.illabout@gmail.com
+Stability   : experimental
+Portability : POSIX
+
+= validate
+
+It is common for passwords to have a set of requirements.
+(e.g. Password must contain at least eight characters that consist of alphabetic
+characters combined with numbers or special characters.)
+
+Validate module provides set of functions which enables you to validate them.
+
+-}
+
+module Data.Password.Validate
+  ( -- * Data type
+    PasswordPolicy (..),
+    InvalidReason (..),
+    CharacterCategory(..),
+    -- * Default
+    defaultPasswordPolicy,
+    -- * Functions
+    valid,
+    validatePassword,
+    -- * Utility
+    isValidPasswordPolicy,
+    defaultCharSet,
+  ) where
+
+import           Data.Char              (isDigit, isLower, isUpper)
+import           Data.Maybe             (catMaybes, fromMaybe)
+import           Data.Password.Internal (Password (..))
+#if! MIN_VERSION_base(4,13,0)
+import           Data.Semigroup         ((<>))
+#endif
+import           Data.Text              (Text)
+import qualified Data.Text              as T
+import           Test.Tasty.QuickCheck  (Arbitrary (..), Gen, choose, elements,
+                                         oneof)
+
+-- $setup
+-- >>> :set -XOverloadedStrings
+--
+-- Import needed libraries.
+--
+-- >>> import Data.Password
+
+-- | Set of policies used to validate 'Password'
+data PasswordPolicy = PasswordPolicy
```

Comment:
I'd prefer the fields not have a prefix in this case. (e.g. instead of `passPolicyMinimumLength`, make it `minimumLength`)
Without the prefix, the chances of there being overlapping definitions with other libraries is fairly small, and users can always import qualified.

`uppercaseChars`, `lowercaseChars`, `specialChars` and `digitChars` might be more intuitive when defining the policy:
`defaultPolicy { uppercaseChars = 3, lowercaseChars = 3, digitChars = 4 }`